instance/tasks.py

```diff
@@ -76,47 +74,81 @@ def start_deployment(
     logger.info('Retrieving deployment: ID=%s', instance_ref_id)
     deployment = OpenEdXDeployment.objects.get(pk=deployment_id)
 
+    # For redundancy - if app server deployment was successful even after cancelling, we prevent the change in
+    # active appserver here
```

Comment:
@Kelketek, in [this comment](https://github.com/open-craft/opencraft/pull/628/files#r469862596) I was referring to moving this comment to the condition in the `check_deactivation()` function where this is relevant.

Here there is no change to the active appserver because the deployment has not yet been started.

Reply:
The reason for grabbing this data during deployment was to avoid any chance of race conditions changing what's there. While it's unlikely to happen, it is possible for someone to change how many app servers are intended to be in an instance before one of the servers has been created. It's also possible for one server to not have started while another one has activated. This could cause data inconsistencies, so I just made the checks all at the beginning.

In unit tests, this problem will almost certainly happen because Huey is always eager, so one server will certainly activate before the rest and the old IDs will  be out of date.

Reply:
@Kelketek, that makes sense. I will revert my changes to the PR and we can go ahead with the merge and deployment after the checks pass.